src/stories/Checkbox.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+
+interface ICheckbox {
+  /*
+  Whether the checkbox is checked.
+  */
+  isChecked?: boolean;
+  /*
+  The text to display next to the checkbox.
+  */
+  text?: string;
+}
+
+// Clicking the checkbox is not updating the initial isChecked value;
+// Can you update storybook controls by only interacting with the component?
+export const Checkbox = ({
+  isChecked = false,
+  text = "",
+  ...props
+}: ICheckbox) => {
+  const [checked, setChecked] = useState<boolean>(isChecked);
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setChecked(event.target.checked);
+  }
+
+  return (
+    <div className="py-2">
+      <input
+        onChange={handleChange}
+        id="check"
```

Comment:
You'll always want `id` to be unique. Have it take in something like `text` or generate some unique value to differentiate it from something else. This is especially important for something like input fields since if you have multiple inputs sharing an ID, the `htmlFor` will target the first input that matches that `ID`, which may not necessarily be the correct one.

Reply:
Ahh, that makes sense, I didn't test a situation with multiple checkboxes... I think I'll change it to `text` for now, but I assume that generating a random `id` would be preferable (is there a recommended way to do that?) since there could be multiple checkboxes on a page with the same label text (most likely two checkboxes with an empty `""` label).

Reply:
Truthfully, I'd suggest having it take in a required `id` prop that the users handles when adding the component. The ID only needs to be unique for the page it's on, and generally speaking people will use something like the `index` for the mapped fields to create unique ID's. Something like `id="input-${index}"` could work. There also is a [`crypto.randomUUID()` method](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) you could use to generate a unique key. This actually is recommended when dealing with `keys` in react. There are possible issues you could run into using an `index` for generating unique keys which is what often is done, but generating a unique ID resolves this problem.